source/jormungandr/jormungandr/scenarios/helper_classes/streetnetwork_path.py

```diff
@@ -80,27 +93,115 @@ def __init__(
         self._request_id = request_id
         self._async_request()
 
-    @new_relic.distributedEvent("direct_path", "street_network")
-    def _direct_path_with_fp(self):
-        with timed_logger(self._logger, 'direct_path_calling_external_service', self._request_id):
-            try:
-                return self._streetnetwork_service.direct_path_with_fp(
+    @staticmethod
+    def poi_to_pt_object(poi):
+        return type_pb2.PtObject(poi=poi, uri=poi.uri, embedded_type=type_pb2.POI, name=poi.name)
+
+    def get_pt_objects(self, entry_point):
+        if include_poi_access_points(self._request, entry_point, self._mode):
+            for o in entry_point.poi.children:
+                yield self.poi_to_pt_object(o)
+        else:
+            yield entry_point
+
+    def make_poi_access_points(self, fallback_type, best_dp_element):
+        entry_point = (
+            self._orig_obj if fallback_type == StreetNetworkPathType.BEGINNING_FALLBACK else self._dest_obj
+        )
+        if not include_poi_access_points(self._request, entry_point, self._mode):
+            return
+        via_poi_access = (
+            best_dp_element.origin
+            if fallback_type == StreetNetworkPathType.BEGINNING_FALLBACK
+            else best_dp_element.destination
+        )
+        for journey in best_dp_element.response.journeys:
+            sections = journey.sections
+            add_poi_access_point(fallback_type, via_poi_access, sections)
+            sections[0].origin.CopyFrom(
+                self._orig_obj
+            ) if fallback_type == StreetNetworkPathType.BEGINNING_FALLBACK else sections[
+                -1
+            ].destination.CopyFrom(
+                self._dest_obj
+            )
+        extend_with_via_poi_access(
```

Comment:
minor remark : it is a bit hard to guess the difference between `add_poi_access_point` and `extend_with_via_poi_access`. At first, I though the same thing was done twice.
Maybe rename as `add_poi_access_point_in_sections` and `extend_path_with_via_poi_access` or something like that ?

Reply:
Done